src/client/configuration.rs

```diff
@@ -102,7 +103,45 @@ impl Configuration {
         })
     }
 
-    pub fn get_feature(&self, feature_id: &str) -> Result<FeatureSnapshot> {
+    /// Returns a mapping of segment ID to `Segment` for all segments referenced
+    /// by the given `segment_rules`.
+    fn get_segments_for_segment_rules(
+        segments: &[Segment],
+        segment_rules: &[TargetingRule],
+    ) -> HashMap<String, Segment> {
+        let referenced_segment_ids = segment_rules
+            .iter()
+            .flat_map(|targeting_rule| {
+                targeting_rule
+                    .rules
+                    .iter()
+                    .flat_map(|segment| &segment.segments)
+            })
+            .cloned()
+            .collect::<HashSet<String>>();
+
+        segments
+            .iter()
+            .filter(|&segment| referenced_segment_ids.contains(&segment.segment_id))
+            .map(|segment| (segment.segment_id.clone(), segment.clone()))
+            .collect()
+    }
+
+    pub(crate) fn get_feature_ids_refs(&self) -> Vec<&String> {
+        self.features.keys().collect()
+    }
+
+    pub(crate) fn get_property_ids_refs(&self) -> Vec<&String> {
+        self.properties.keys().collect()
+    }
+}
```

Comment:
Nothing has changed here, it's just moved before the `ConfigurationProvider` implementation